airflow/providers/amazon/aws/sensors/athena.py

```diff
@@ -78,7 +78,11 @@ def poke(self, context: Context) -> bool:
         state = self.hook.poll_query_status(self.query_execution_id, self.max_retries, self.sleep_time)
 
         if state in self.FAILURE_STATES:
-            raise AirflowException("Athena sensor failed")
+            # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
```

Comment:
Just out of curiosity. Why? Why we no longer need this check from 2.7.1?

Reply:
@vincbeck Here - https://github.com/apache/airflow/pull/33405#discussion_r1304722753